### ISSUE TITLE
Liquid profile not found for this page

### ISSUE THREAD

**Issue by henrybarn** [Shopify/shopify-theme-inspector#34]
### Problem

I've tried two Shopify stores with a single-login full access account and both times received the message 'This page cannot be profiled'

The console gives me one error:
```
devtools.js:16 Error: Liquid profile not found for this page
```

### Replication steps

I'm not sure but I'm glad to provide more information. I'm possibly not satisfying all of the requirements for this tool

### More Information

The XHR for example.com/collections/all?profile_liquid=true doesn't respond with anything different from a request to example.com/collections/all

**Comment by kieranmasterton** [comment]
Same issue here on the three stores I've tested. I've tried on Chrome on Ubuntu and Mac OS. In the inspector I can see the request to `/?profile_liquid=true` succeeds but then the console reports the following error: 

`Error: Liquid profile not found for this page`

Shop IDs tested: `11829138`, `11829138` and `19736725`

![Screenshot from 2020-01-15 19-38-50](https://user-images.githubusercontent.com/90990/72465652-7c37a180-37cf-11ea-8ddf-a458ee951fb6.png)

**Comment by henrybarn** [comment]
Looks like I got it to work for some shops on their `.myshopify.com` domain. I think the issue might be related to shops that are on a custom domain name. I'm not yet sure how to bypass the redirection rules.

**Comment by kieranmasterton** [comment]
That's confirmed, I've just switched the primary domain to the `myshopify.com` domain for one of my test stores that failed previously and it now works. Looks like say you said @henrybarn, it's failing for stores with custom domains.

**Comment by kieranmasterton** [comment]
@henrybarn you can click `Disable redirection` in the Admin under Online Store > Domains to get access via the `myshopify.com` domain but it's obviously not possible if you have other domains you're redirecting to the primary domain.

**Comment by lukechadwick** [comment]
I'm having this issue on both my test store with a myshoify URL and a Plus store with custom domain. It profiles the homepage and product pages fine, but the I get the above error on the cart page.

**Comment by tehaksbrid** [comment]
Same. We redirect from our shopify url so I can't comment on whether or not running from there works. I would guess that authentication is strongly coupled to the domain.

Digging into their extensions, it's _looking_ like Shopify is supposed to be injecting an element with id `liquidProfileData` in response to that query parameter. The element is absent, which is throwing the error we're all seeing in console.

An important looking authentication-related promise is also rejecting with response `{}`, namely one for:
```
chrome.runtime.sendMessage({
                        type: "request-core-access-token",
                        origin: t
                    }, ({token: t, error: r})=>r ? n(r) : e(t))
```

#33 is probably the same issue

**Comment by tehaksbrid** [comment]
After disabling ***.myshopify.com -> custom-domain.com redirecting for the store, the profiler worked. You can also try the other fixes suggested in #33
